core-java/src/main/proto/spine/money/money.proto

```diff
@@ -0,0 +1,408 @@
+//
+// Copyright 2015, TeamDev Ltd. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+syntax = "proto3";
+
+package spine.money;
+
+option java_generate_equals_and_hash = true;
+option java_multiple_files = true;
+option java_outer_classname = "MoneyProto";
+option java_package = "org.spine3.money";
+
+import "google/protobuf/descriptor.proto";
+
+// Represents the amount of money.
+message Money {
+    // Represents the amount of money (e.g. cents).
+    sint64 amount = 1;
+
+    // Represents the currency.
+    Currency currency = 2;
+}
+
+message CurrencyOptions {
+    int32 numeric_code = 1;
```

Comment:
Please extend options with all the fields from the table at https://www.wikiwand.com/en/ISO_4217.
We already have the name as the item of the enum. We've got `numeric_code`.
We definitely need `exponent_digits`, as it would serve as the basis for interpreting the `amount` field.

We need to store the default name of the currency somewhere. Why not have it as the `name` or `title` field? Localization to other languages would use this value as a key for language-specific versions.

Information in the `country` field (assuming it can be repeated), would help in listing default currency basing on a geo location of the user.